sql/core/src/main/scala/org/apache/spark/sql/execution/aggregate/utils.scala

```diff
@@ -343,52 +348,41 @@ object Utils {
         (rewrittenAggregateExpression -> aggregateFunctionAttribute)
     }.unzip
 
-    val finalAndCompleteAggregate = if (usesTungstenAggregate) {
-      val rewrittenResultExpressions = resultExpressions.map { expr =>
-        expr.transform {
-          case agg: AggregateExpression2 =>
-            val function = agg.aggregateFunction
-            val isDistinct = agg.isDistinct
-            val aggregateFunction =
-              if (rewrittenAggregateFunctions.contains(function, isDistinct)) {
-                // If this function has been rewritten, we get the rewritten version from
-                // rewrittenAggregateFunctions.
-                rewrittenAggregateFunctions(function, isDistinct)
-              } else {
-                // Oterwise, we get it from aggregateFunctionMap, which contains unique
-                // aggregate functions that have not been rewritten.
-                aggregateFunctionMap(function, isDistinct)._1
-              }
-            aggregateFunction.asInstanceOf[ExpressionAggregate].evaluateExpression
-          case expression =>
-            // We do not rely on the equality check at here since attributes may
-            // different cosmetically. Instead, we use semanticEquals.
-            groupExpressionMap.collectFirst {
-              case (expr, ne) if expr semanticEquals expression => ne.toAttribute
-            }.getOrElse(expression)
-        }.asInstanceOf[NamedExpression]
-      }
+    // The original `resultExpressions` are a set of expressions which may reference aggregate
+    // expressions, grouping column values, and constants. When aggregate operator emits output
+    // rows, we will use `resultExpressions` to generate an output projection which takes the
+    // grouping columns and final aggregate result buffer as input. Thus, we must re-write the
+    // result expressions so that their attributes match up with the attributes of the final
+    // result projection's input row:
+    val rewrittenResultExpressions = resultExpressions.map { expr =>
+      expr.transform {  // TODO(josh): why is this transform instead of transformDown
```

Comment:
@yhuai, I noticed that this case (in the old code, too) used `transform` instead of `transformDown`. Any significance to that?